lib/ecto/adapters/mysql/connection.ex

```diff
@@ -495,7 +495,7 @@ if Code.ensure_loaded?(Mariaex) do
         case elem(sources, pos) do
           {table, schema} ->
             name = String.first(table) <> Integer.to_string(pos)
-            {quote_table(prefix, table), name, schema}
+            {quote_table(prefix || (schema && schema.__schema__(:prefix)), table), name, schema}
```

Comment:
We don't support queries across schemas by design. If we are going to support it, we need to revisit all cases and scenarios, it should not be handled in the adapter because the adapter may not have enough information to decide which one to use.